pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -985,18 +983,36 @@ func (l *LibvirtDomainManager) SyncVMI(vmi *v1.VirtualMachineInstance, allowEmul
 		}
 	}
 
-	ifaceNamingScheme := namescheme.CreateNetworkNameScheme(vmi.Spec.Networks)
+	vmConfigurator := netsetup.NewVMNetworkConfigurator(vmi, cache.CacheCreator{})
+	indexedInterfaces := netvmispec.IndexInterfaceSpecByName(vmi.Spec.Domain.Devices.Interfaces)
 	for _, network := range netvmispec.NetworksToHotplugWhosePodIfacesAreReady(vmi) {
 		log.Log.Infof("will hot plug %s", network.Name)
-		podIfaceName, wasFound := ifaceNamingScheme[network.Name]
+
+		ifaceToHotplug, wasFound := indexedInterfaces[network.Name]
 		if !wasFound {
-			return nil, fmt.Errorf("could not find the pod interface name for network [%s]", network.Name)
+			log.Log.Errorf("could not find a matching interface for network: %s", network.Name)
+			return nil, err
 		}
 
-		ifaceXML, err := xml.Marshal(converter.DomainInterfaceSpec(c, network.Name, virtnetlink.GenerateTapDeviceName(podIfaceName)))
+		domainInterfaces, err := converter.CreateDomainInterfaces(vmi, domain, c, converter.IsVirtioNetProhibited(), ifaceToHotplug)
```

Comment:
I'm a bit confused.
For me, the trivial patch would have been to just move the hotplug thing into the `CreateDomainInterfaces` or better, define a differentiator that compares between that exists and what is desired, and returns only what needs to be hotplug (like [this](https://github.com/kubevirt/kubevirt/blob/5588ee3a969d0ecd433b79d59bcd61ccb171e05a/pkg/virt-launcher/virtwrap/device/hostdevice/hotplug.go#L182)).

Reply:
The confusion is that the code iterates over each network that needs to be plugged, then calls `CreateDomainInterfaces` to regenerate *all* needed interfaces again.
And `domain.Spec.Devices.Interfaces` is overridden each time.

Hope this is clear now.

Reply:
Thank you. Now I understand.

I'll see what I can do.

Reply:
Is this better now ? I am now appending to the current domain interface slice instead of overriding it.

Create domain interfaces will handle only the interface you pass it when provided variadic arguments.